documentation/binding-data/tutorial-flow-data-provider.asciidoc

```diff
@@ -272,28 +272,45 @@ The lazy loaded binding in `ComboBox` is always filtered by the string typed
 in by the end user.
 Initially, when there is no filter input yet, the filter is an empty string.
 
-// TODO change this into data view style once the changes are in ComboBox
-
-`ComboBox` currently uses the old data provider naming convention.
-Also, the count of total matches is currently required.
+*The `ComboBox` examples listed below use an API available only in Vaadin 18+.*
 
 For example, you can handle filterable lazy data binding to a Spring Data
 repository as follows:
 
 [source, java]
 ----
 ComboBox<Person> cb = new ComboBox<>();
-cb.setDataProvider((String filter, int offset, int limit) -> {
-    return repo.findByNameLikeIgnoreCase(
-            "%" + filter + "%", // <1>
-            PageRequest.of(offset / limit, limit)
-    ).stream();
-}, filter -> {
-    return (int) repo.countByNameLikeIgnoreCase("%" + filter + "%"); // <2>
-});
+cb.setItems(
+        query -> repo.findByNameLikeIgnoreCase( // <1>
+                "%" + query.getFilter().orElse("") + "%", // <2>
+                PageRequest.of(query.getPage(), query.getPageSize()))
+                .stream()
+);
 ----
-<1> Add `%` marks to filter for an SQL "LIKE" query.
-<2> The total amount of items matching the filter is required.
+<1> Here we use only the fetch callback for lazy loading with undefined item count. The item count callback is optional.
+<2> Add `%` marks to filter for an SQL "LIKE" query.
+Use it if getting the count is cheap and/or a real scroll bar size is desired.
+
+You may want to filter the items in `ComboBox` by some value type other than a string.
+If your backend is able to fetch the items by this value, you can set the filter converter along with fetch callback.
```

Comment:
Again, I would instead make it passive:
"In case the filtering of items should be done with another type than string, you can provide a filter converter with the fetch callback to get the type of the filter right for the fetch query. ...:"

Reply:
done